File: 9-palindrome-number/9-palindrome-number.scala
object Solution {
  def isPalindrome(x: Int): Boolean = {
    if (x < 0 || (x%10==0 && x!=0)) return false
    var org = x
    var rev = 0
    while (org > rev) {
      rev = rev * 10 + org % 10
      org /= 10
    }
    return org == rev || rev /10 == org
  }
}

File: 9-palindrome-number/9-palindrome-number.go
func isPalindrome(x int) bool {
  if x < 0 { return false }
  if x < 10 { return true }
  rev := 0
  for rem := x; rem > 0; rem /= 10 {
    rev = rev * 10 + rem % 10
  }
  return rev == x
}